### ISSUE TITLE
User Feedback required: UI/UX Challenges in Peak curation

### ISSUE THREAD

**Issue by shubhra-agrawal** [ElucidataInc/ElMaven#868]
We have been reviewing El-MAVEN with respect to the user experience and UI. Some major challenges that were identified have been listed below. We would like to get some feedback on this list. Let us know if there's something we're missing or if there's something you disagree with.

- Report isotopic check is hidden in the Peaks dialog
- Changing global parameters changes the EIC of curated peaks
- User cannot go back to the settings used for a particular table
- Comparing multiple groups for a metabolite in one view
- User does not see the instrumentation and file import settings before loading samples. Has to reload the samples eventually.
- Overhead of merging bookmark and peak table
- Too many clicks from metabolite to visualizing label EIC
- Overhead of moving between compound db and peak table during curation
- No defined workflow(s)
- Workflow-dependent parameters are not well-distinguished from the post-analysis visualisations
- UI gets very cluttered in case of multiple peak tables

@chubukov @lparsons @shefalilathwal @raaisakuk @sunil20dhakad @eugenemel

**Comment by raaisakuk** [comment]
@shubhra-agrawal When you say "No defined workflows", what do you exactly mean by that? Also, could you include https://github.com/ElucidataInc/ElMaven/issues/625 in the list?

**Comment by chubukov** [comment]
If I understand correctly what is meant by a "workflow", the idea is to basically guide a user who does not have extensive experience looking at LCMS data through some common steps.

Perhaps there is some room for that, but I would not make it a priority -- I don't necessarily want to turn maven into a product like Progenesis, where all the details are hidden and using it almost feels like you're supposed to just keep pressing "Next".

**Comment by chubukov** [comment]
In terms of similar entries, I would add that it can be very annoying to deal with selecting/deselecting samples.

- when viewing a messy EIC with a lot of samples, I often want to be able to just see one sample, so that I can have a clearer idea of e.g. if there are two peaks in a single sample or not. This requires a lot of clicks currently, both to see it, and then to reset the state back.
- even within the sample widget, if you some samples selected and some deselected, getting to "all selected" can take a lot of clicks.

**Comment by chubukov** [comment]
Otherwise, I agree with many of the comments that have to do with it being sometimes very tricky to understand what parameters were fixed when a group was created, and which ones are still being changed in the main window.

**Comment by raaisakuk** [comment]
I agree about the selecting/delecting... One of the solutions I proposed at some point was https://github.com/ElucidataInc/ElMaven/issues/569
It might be worth checking out now.

**Comment by shubhra-agrawal** [comment]
@chubukov @raaisakuk #569 is a good option for looking at a single sample at a time. Might need some tweaks to make it work for certain sets of samples at a time, though. We'll take that into consideration while planning the UX. I'll also add this challenge to the list above.

**Comment by shubhra-agrawal** [comment]
> Perhaps there is some room for that, but I would not make it a priority -- I don't necessarily want to turn maven into a product like Progenesis, where all the details are hidden and using it almost feels like you're supposed to just keep pressing "Next".

@chubukov I agree. Any changes we make for a better UX won't come at the cost of functionality. All necessary parameters will still be available to the user-- front and center. All we want to do is put those options in context. For eg. the import and file instrumentation options should be available before sample upload but those are hidden in the Options dialog. More often than not, users have to set those parameters, close El-MAVEN and then reload their samples.

We would like to do away with some of these ambiguities. Separating the targeted and untargeted workflows is one of the efforts around this.

We have come up with some designs that can help overcome most of these challenges listed above. These will be shared with the users for discussion once we have incorporated more of your suggestions and issues. Hope to share them with you soon!

**Comment by shefalilathwal** [comment]
@shubhra-agrawal I want to add that I often also get confused between where all the parameters are. For example, group filtering parameters are in the peaks dialog box, whereas peak filtering parameters and baseline parameters are in the options dialog box. Same thing with isotopic detection parameters, they are scattered in multiple places.

**Comment by surbhipoddar** [comment]
@shefalilathwal I agree. All the parameters should appear in context as and when they are required in the workflow. 

As mentioned, like the instrumentation and file import parameters, we are planning to place all the parameters regarding group filtering and peak picking in context to your analysis.

**Comment by chubukov** [comment]
> @chubukov @raaisakuk #569 is a good option for looking at a single sample at a time. Might need some tweaks to make it work for certain sets of samples at a time, though. We'll take that into consideration while planning the UX. I'll also add this challenge to the list above.
> @surbhipoddar FYI

I was thinking about something like: be able to right-click on a peak in the EIC window and have the option "show just this sample". And then a one-click way to reset the view back to what it was.

**Comment by surbhipoddar** [comment]
> > @chubukov @raaisakuk #569 is a good option for looking at a single sample at a time. Might need some tweaks to make it work for certain sets of samples at a time, though. We'll take that into consideration while planning the UX. I'll also add this challenge to the list above.
> > @surbhipoddar FYI
> 
> I was thinking about something like: be able to right-click on a peak in the EIC window and have the option "show just this sample". And then a one-click way to reset the view back to what it was.
> 
> Edit: or, even cooler but way more work: allow clicking on a particular peak to bring up just that sample's EIC in a new window/widget. And you could imagine then being able to set sample-specific bounds on the peak, change baseline, etc in this window. Not saying that this is high on my list, just an idea of where this could go.

@chubukov These ideas sound really nice. Implementing this is going to require some insight from @shubhra-agrawal since currently, we look at groups in the EIC rather than single samples.

**Comment by lparsons** [comment]
I'm not sure this is the appropriate place to mention it, but I'm getting an increasing amount of feedback related to sample ordering (issue #913).

**Comment by lparsons** [comment]
Also, a very common complaint is instability (El-Maven crashes far to frequently). It would be helpful if you had a list of crash reports associated with our lab members. I'd like to at least be able to track the issues and may need to prioritize some fixes.

**Comment by rish9511** [comment]
@lparsons 

Making Crash reports and Sentry integration better is on top of our TO-DO list after we are done working on the release. 

> It would be helpful if you had a list of crash reports associated with our lab members. I'd like to at least be able to track the issues and may need to prioritize some fixes.

I believe the effort we are going to put to make our Sentry integration better will also incorporate the changes that you have requested above. 
We will keep you in loop when we start working on it 

> Also, a very common complaint is instability (El-Maven crashes far to frequently)

Is there something that constantly crashes for you? Asking because we have noticed less crashes over the last few months. And is it something that's easily reproducible ?

**Comment by lparsons** [comment]
@rish9511 I'm working on getting further details on the crashes and will report back here once I get some information on reproducibility, etc.

**Comment by lparsons** [comment]
Another pain point relates to the isotope plot. It doesn't update unless the user clicks directly on the peak (unlike the isotope widget). This can significantly slow down workflow. See #703.

**Comment by lparsons** [comment]
@rish9511 I was able to "reproduce" the crash the user reported, though it took a few tries. The issue happened after changing the `ppm` and then clicking on a peak. The crash report was just sent (~10:47 EST, 2018-11-14). I submitted issue #918 to track it.

**Comment by lparsons** [comment]
Another particularly annoying UX bug: #820 - Window size and geometry is not saved between sessions. This makes it quite difficult to get widgets setup to optimize workflow.

**Comment by shubhra-agrawal** [comment]
Adding another UX issue from the #453 thread:
The EIC widget loads back the last viewed slice from the previous session as soon as you load sample files. This is not useful and only adds to ambiguity when it comes to interacting with those peaks. I suggest making the EIC widget blank for a new session unless the user chooses to look at a peak.
